server/message_hooks_test.go

```diff
@@ -1995,3 +1995,177 @@ func TestGetChatIDForChannel(t *testing.T) {
+			ExpectedMessage: "Hi <at id=\"0\">@all</at>",
+			ChatID:          testutils.GetChatID(),
+			SetupAPI: func(api *plugintest.API) {
+				api.On("LogDebug", "Unable to get ms teams chat", "Error", "error occurred while getting chat")
+			},
+			SetupClient: func(client *clientmocks.Client) {
+				client.On("GetChat", testutils.GetChatID()).Return(nil, errors.New("error occurred while getting chat"))
+			},
+			SetupStore:            func(store *storemocks.Store) {},
+			ExpectedMentionsCount: 1,
+		},
+		{
+			Name:            "GetMentionsData: mentioned all in Teams channel",
+			Message:         "Hi @all",
+			ExpectedMessage: "Hi <at id=\"0\">mock-name</at>",
+			SetupAPI:        func(api *plugintest.API) {},
+			SetupClient: func(client *clientmocks.Client) {
+				client.On("GetChannelInTeam", testutils.GetTeamID(), testutils.GetChannelID()).Return(&msteams.Channel{
+					DisplayName: "mock-name",
+				}, nil)
+			},
+			SetupStore:            func(store *storemocks.Store) {},
+			ExpectedMentionsCount: 1,
+		},
+		{
+			Name:            "GetMentionsData: error occurred while getting the MS Teams channel",
+			Message:         "Hi @all",
+			ExpectedMessage: "Hi <at id=\"0\">@all</at>",
+			SetupAPI: func(api *plugintest.API) {
+				api.On("LogDebug", "Unable to get ms teams channel", "Error", "error occurred while getting the MS Teams channel")
+			},
+			SetupClient: func(client *clientmocks.Client) {
+				client.On("GetChannelInTeam", testutils.GetTeamID(), testutils.GetChannelID()).Return(nil, errors.New("error occurred while getting the MS Teams channel"))
+			},
+			SetupStore:            func(store *storemocks.Store) {},
+			ExpectedMentionsCount: 1,
+		},
+		{
+			Name:            "GetMentionsData: mentioned a user",
+			Message:         "Hi @test-username",
+			ExpectedMessage: "Hi <at id=\"0\">mock-name</at>",
+			ChatID:          testutils.GetChatID(),
+			SetupAPI: func(api *plugintest.API) {
+				api.On("GetUserByUsername", "test-username").Return(testutils.GetUser("mock-role", "mock-email"), nil)
+			},
+			SetupClient: func(client *clientmocks.Client) {
+				client.On("GetUser", testutils.GetUserID()).Return(&msteams.User{
+					DisplayName: "mock-name",
+				}, nil)
+			},
+			SetupStore: func(store *storemocks.Store) {
+				store.On("MattermostToTeamsUserID", testutils.GetID()).Return(testutils.GetUserID(), nil)
+			},
+			ExpectedMentionsCount: 1,
+		},
+		{
+			Name:            "GetMentionsData: mentioned all and a specific user in a channel",
+			Message:         "Hi @all @test-username",
+			ExpectedMessage: "Hi <at id=\"0\">Everyone</at> <at id=\"1\">mock-name</at>",
+			ChatID:          testutils.GetChatID(),
+			SetupAPI: func(api *plugintest.API) {
+				api.On("GetUserByUsername", "test-username").Return(testutils.GetUser("mock-role", "mock-email"), nil)
+			},
+			SetupClient: func(client *clientmocks.Client) {
+				client.On("GetChat", testutils.GetChatID()).Return(&msteams.Chat{
+					Type: "G",
```

Comment:
check if there is a constant present for this

Reply:
Not present